### DISCUSSION TITLE
Unfortunate update crashed entire Azuracast installation.

### DISCUSSION THREAD

**Discussion by radiobijoux** [AzuraCast/AzuraCast#4371] [General]
Sadly, today after the most recent rolling update, the station began dropping the stream after approximately 24 seconds into the broadcast connection.  This happened every single time.  I attempted virtually everything I could to determine the cause, but with no success.  So I went back to my system backups and selected the most recent from last evening's midnight backup.  After following the directions, I eventually was able to restore the backup.  Upon completion, the site reloaded to the New Station Setup routine and the entire original station information was gone.  I don't know why as I downloaded and extracted the sql file to see if the database had indeed been backed up, and it had.  This was on my Oracle free tier.  I'm also running a standard deployment on a DigitalOcean setup, so I was able to switch streams to my site temporarily (or permanently as the case may be).  I don't know why my install did this or why it didn't restore correctly, and sadly, the database is not really easily accessible through any normal means to easily upload the extracted database, so I'm contemplating just restarting or scrapping the Oracle install.  My advice is to discontinue the use of a rolling install until there is an easier way to recover database.  That might be something that should be implemented within the Azuracast since it seems to control every aspect of the "site" it creates.  Better to be safe than sorry at any rate.

**Comment by ItsMitchh** [comment]
Our rolling release branches are visibly marked as possibly unstable, not fully tested and not a complete stable form, users are informed to make sure they have backed up their installations. While I'm sorry to hear you had issues, without any kind of logs or a GH issue, I got no idea as to the root cause of this.

**Reply by radiobijoux** [reply]
I understand.  I don't have a clue where to begin, so I just wanted to describe the issues I personally had in the hopes that some of the issues could be traced back based to the updates that were included in the release since the original installation ended up being removed after my attempt to restore a backup (no logs or anything to retrieve).   I appreciate the hard work you're putting into this and after I've had the chance to learn more about it, I can be of somewhat more use.   I had also contemplated a potential problem with the rolling releases so I set up the second install using the stable release option just in case, so I had a fallback.

**Comment by BusterNeece** [comment]
@radiobijoux When you update, are you also running `./docker.sh update-self` to update the update script itself? This is a very important step and often fixes issues that people encounter when udpating.

The specific problem you're describing, the stations shutting down rather quickly after launching, does indeed sound like something we encountered many versions ago, but we fixed it within the day. If you haven't done the latter step before updating, it's possible you're "stuck" on that buggy version somehow.

From your message, though, it appears you've already attempted to restore your installation (only to have that not work either). There are ways of manually installing DB dumps into our MariaDB container if necessary. It's a little complicated, but if you have the SQL dump, restoration isn't working (even with the `update-self` command from above) and you need that data, we can help with getting that back into the database.

**Reply by radiobijoux** [reply]
Thanks for the response.  Yes, I updated using the update procedure spelled out in the Docs.  The update-self was performed prior to the update.  To give you a little more information.  I set up a free lifetime tier VPS on Cloud Oracle for testing purposes.  Their free tier level provides you with 1 GB memory, which is lower than the recommended.  The information I reviewed and used was to create a 1 GB swap file to provide more flexibility for the docker/Azuracast installation.  I am now wondering if during the update process, somehow the update got corrupted due to this fact?  The rest of the "hardware" is fairly generic and meets the minimum requirements.  What I DID notice during the attempted restore of the database was that a number of tables were "missing".  I cannot remember which, but there were about 4 instances of database tables "missing".  So I am wondering if something was changed during the update that the prior backup did not contain in the database?  I wish I could provide you with more information, and due to the way the Oracle "instance" has to be accessed using an SSH Key Pair, if you get the time and can provide me with the instructions for dumping the SQL backup into the database, I can handle that.  I've still got my DigitalOcean installation and am working from that since it has the minimum requirements for everything.  Thanks!  I appreciate all the hard work you've put into this and how much money you're saving me and my nonprofit by providing it.
